express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const notesJsonData = require('./data.json');
+let notes = Object.values(notesJsonData.notes);
+let nextId = notesJsonData.nextId;
+
+const jsonMiddleware = express.json();
+app.use(jsonMiddleware);
+
+app.get('/api/notes', (req, res, next) => {
+  res.status(200).json(notes);
+});
+
+app.get('/api/notes/:id', (req, res, next) => {
+  const reqId = parseInt(req.params.id, 10);
+  const note = notes.filter(x => x.id === reqId);
+  note.length
+    ? res.status(200).json(note[0])
+    : Math.floor(reqId) !== reqId || typeof reqId !== 'number'
+      ? res.status(400).json({ error: 'id must be a positive integer' })
+      : res.status(404).json({ error: 'cannot find note with id ' + reqId });
+});
+
+app.post('/api/notes', (req, res, next) => {
+  if (!req.body.content) {
+    res.status(400).json({ error: 'content is a required field' });
+  } else if (typeof req.body === 'undefined') {
+    res.status(500).json({ error: 'An unexpected error occurred.' });
```

Comment:
This condition cannot happen, because you are using the `express.json()` middleware. The `500` case is if there is an error writing the data to the file
```js
fs.writeFile('wrong-path/data.json', err => {
  if (err) {
    res.status(500).json({ error: 'an unexpected error occurred' })
  } else {
    res.status(201).json(theNewNoteObjectWithId)
  }
})
```

Reply:
This problem should be fixed in all of your routes.